pkg/plugins/resources/postgres/store_template_test.go

```diff
@@ -4,29 +4,43 @@ import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
+	"github.com/kumahq/kuma/pkg/config/plugins/resources/postgres"
 	"github.com/kumahq/kuma/pkg/core/resources/store"
 	core_metrics "github.com/kumahq/kuma/pkg/metrics"
 	test_store "github.com/kumahq/kuma/pkg/test/store"
 	test_postgres "github.com/kumahq/kuma/pkg/test/store/postgres"
 )
 
 var _ = Describe("PostgresStore template", func() {
-	createStore := func() store.ResourceStore {
+	createStore := func(storeName string) store.ResourceStore {
 		cfg, err := c.Config(test_postgres.WithRandomDb)
 		Expect(err).ToNot(HaveOccurred())
+		cfg.MaxOpenConnections = 2
 
-		metrics, err := core_metrics.NewMetrics("Standalone")
+		pqMetrics, err := core_metrics.NewMetrics("Standalone")
+		Expect(err).ToNot(HaveOccurred())
+
+		pgxMetrics, err := core_metrics.NewMetrics("Standalone")
 		Expect(err).ToNot(HaveOccurred())
 
 		_, err = MigrateDb(*cfg)
 		Expect(err).ToNot(HaveOccurred())
 
-		pStore, err := NewStore(metrics, *cfg)
+		var pStore store.ResourceStore
+		if storeName == "pgx" {
+			cfg.DriverName = postgres.DriverNamePgx
+			pStore, err = NewPgxStore(pgxMetrics, *cfg)
+		} else {
+			cfg.DriverName = postgres.DriverNamePq
+			pStore, err = NewPqStore(pqMetrics, *cfg)
+		}
 		Expect(err).ToNot(HaveOccurred())
 
 		return pStore
 	}
 
-	test_store.ExecuteStoreTests(createStore)
-	test_store.ExecuteOwnerTests(createStore)
+	test_store.ExecuteStoreTests(createStore, "pgx")
```

Comment:
It's kind of strange that `ExecuteStoreTest` will call `createStore` with this argument and this is implemented just for this case (or am I wrong?). Could we solve this with higher order function?

something like this
```
	createStore := func(driverName string) func() store.ResourceStore {
		return func() store.ResourceStore {
			cfg, err := c.Config(test_postgres.WithRandomDb)
			Expect(err).ToNot(HaveOccurred())
			cfg.MaxOpenConnections = 2

			pqMetrics, err := core_metrics.NewMetrics("Standalone")
			Expect(err).ToNot(HaveOccurred())

			pgxMetrics, err := core_metrics.NewMetrics("Standalone")
			Expect(err).ToNot(HaveOccurred())

			_, err = MigrateDb(*cfg)
			Expect(err).ToNot(HaveOccurred())

			var pStore store.ResourceStore
			if storeName == "pgx" {
				cfg.DriverName = postgres.DriverNamePgx
				pStore, err = NewPgxStore(pgxMetrics, *cfg)
			} else {
				cfg.DriverName = postgres.DriverNamePq
				pStore, err = NewPqStore(pqMetrics, *cfg)
			}
			Expect(err).ToNot(HaveOccurred())

			return pStore
		}
	}

	test_store.ExecuteStoreTests(createStore("pgx"))
```

Reply:
passing the name is ok, passing the name and then passing it to createStore func where kubernetes and in-memory ignores it is kind of strange

Reply:
Ok, the problem wasn't the context, I should've included the whole function inside `createStore` function. Now that I changed it - it works. Let me know if this is ok now.